File: 2018/March/March-11/March-11-2018.cpp
//***************************
//*  191. Number of 1 Bits  *
//***************************

//O(1) solution [Because n is a 32-bit integer, and that's what is looped through]
class Solution 
{
public:
    int hammingWeight(uint32_t n) 
    {
        int count = 0;
        uint32_t check = 0;
        
        while (n > 0)
        {
            check = n & 1;
            if (check == 1)
            {
                count++;
            }
            
            n >>= 1;
        }
        
        return count;
    }
};

File: 2018/March/March-11/March-11-2018.java
//***************************
//*  191. Number of 1 Bits  *
//***************************

//O(1) solution [Because n is a 32-bit integer, and that's what is looped through]
public class Solution 
{
    public int hammingWeight(int n) 
    {
        int count = 0;
        long check = 0;
        long m = n;
        int i = 0;
        
        while(i < 32)
        {
            check = m & 1;
            if(check==1)
            {
                count += 1;
            }
            m >>= 1;
            i++;
        }
        
        return count;
    }
}